app/Services/QueryServiceProvider.js

```diff
@@ -37,6 +37,60 @@ async function getUnresolvedVocabulary(languagePackageId, userId) {
   return number;
 }
 
+// return the number of unresolved vocabulary
+async function getQueryVocabulary(languagePackageId, userId, limit) {
+  // Get drawers belonging to languagePackage
+  const drawers = await Drawer.findAll({
+    attributes: ['id', 'name', 'queryInterval'],
+    where: {
+      userId: userId,
+      languagePackageId: languagePackageId,
+    },
+  });
+
+  let vocabs = [];
+
+  for (let key = 0; key < drawers.length; ++key) {
+    let vocabularyLimit = limit;
+    // subtract size of vocabs returned to update the limit
+    vocabularyLimit -= vocabs.length;
+    // create date and add days from query Interval
+    let queryDate = new Date();
+    // subtract query interval from actual date
+    queryDate.setDate(queryDate.getDate() - drawers[key].queryInterval);
+
+    // compare query date with with last query
+    // if queryDate is less than lastQuery: still time
+    // if queryDate more than lastQuery: waiting time is over
+    const vocabularies = await VocabularyCard.findAll({
+      limit: vocabularyLimit,
+      attributes: ['id', 'name'],
+      where: {
+        drawerId: drawers[key].id,
+        lastQuery: { lt: queryDate },
```

Comment:
Does this work?

Reply:
If you mean comparing the two dates, this works absolutely fine.

Reply:
Ok.